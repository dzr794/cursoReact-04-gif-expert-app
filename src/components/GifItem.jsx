import PropTypes from 'prop-types'

const GifItem = ({url, title}) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}

// GifItem.defaultProps = {
//   url: 'No URL given',
//   title: "No title given",
// }

export default GifItem;