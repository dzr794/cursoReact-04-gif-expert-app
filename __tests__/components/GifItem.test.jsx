import {render, screen} from '@testing-library/react'

import GifItem from "../../src/components/GifItem";

describe('Testing <GifItem />', ()=>{
  
  //Arrange
  const url = "https://media3.giphy.com/media/nvzTyGTzU7Xkq5ZF1g/giphy.gif?cid=138a3abfkany7ppigeh3xxws8vxieg8ox7oxjeeyvtehmd76&ep=v1_gifs_search&rid=giphy.gif&ct=g";
  const title = 'Mi tÃ­tulo';

  test('Should match Snapshot', ()=>{
    
    //Act
    const { container } = render(<GifItem title={title} url={url} />);

    //Assert
    //? more info in: https://jestjs.io/docs/expect
    expect( container ).toMatchSnapshot();
    
  });

  test('Should show image from URL and alt with title prop', () => { 
    
    //Act
    render(<GifItem title={title} url={url} />);

    const {alt, src} = screen.getByRole('img');
    
    //Assert
    //? more info in: https://jestjs.io/docs/expect
    expect( alt ).toBe( title );
    expect( src ).toBe( url );

  });

  test('Should match title', () => {
    //Act
    render(<GifItem title={title} url={url} />);

    expect( screen.getByText(title) ).toBeTruthy();

  });

  
});