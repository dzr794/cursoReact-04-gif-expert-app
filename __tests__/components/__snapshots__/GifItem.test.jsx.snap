// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> Should match Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Mi título"
      src="https://media3.giphy.com/media/nvzTyGTzU7Xkq5ZF1g/giphy.gif?cid=138a3abfkany7ppigeh3xxws8vxieg8ox7oxjeeyvtehmd76&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Mi título
    </p>
  </div>
</div>
`;
